src/InMemoryEventStore.php

```diff
@@ -314,16 +303,46 @@ public function fetchStreamNames(
         return $result;
     }
 
-    public function fetchCategoryNames(?string $filter, bool $regex, int $limit, int $offset): array
-    {
-        if (null === $filter && $regex) {
-            throw new Exception\InvalidArgumentException('No regex pattern given');
+    public function fetchStreamNamesRegex(
+        string $filter,
+        ?MetadataMatcher $metadataMatcher,
+        int $limit = 20,
+        int $offset = 0
+    ): array {
+        if (false === @preg_match("/$filter/", '')) {
```

Comment:
Ain't it much slower to register & unregister the error handler for that single line of code?